tests/schema/tests.py

```diff
@@ -826,6 +842,96 @@ def test_alter_fk_to_o2o(self):
                     author_is_fk = True
         self.assertTrue(author_is_fk, "No FK constraint for author_id found")
 
+    def test_alter_field_fk_to_o2o(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+            editor.create_model(Book)
+        old_field = Book._meta.get_field('author')
+        new_field = OneToOneField(Author, CASCADE)
+        new_field.set_attributes_from_name('author')
+        with connection.schema_editor() as editor:
+            editor.alter_field(Book, old_field, new_field, strict=True)
+
+        fks, uniques, indexes = self.count_constraints(
+            Book._meta.db_table,
+            Book._meta.get_field('author').column,
+            (Author._meta.db_table, Author._meta.pk.column),
+        )
+
+        if connection.features.supports_foreign_keys:
```

Comment:
It might be useful to make these assertions in tuple form so that if something goes wrong, it's slightly easier to see what might be wrong, e.g. `assertEqual((fks, uniques, indexes), (1, 1, 0))`